test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/SelectExpandClauseExtensionsTests.cs

```diff
@@ -0,0 +1,334 @@
+// </copyright>
+//---------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.OData.Tests.UriParser;
+using Microsoft.OData.UriParser;
+using Xunit;
+
+namespace Microsoft.OData.Tests.ScenarioTests.UriParser
+{
+    public class SelectExpandClauseExtensionsTests
+    {
+        [Fact]
+        public void GetCurrentLevelSelectListTestMultiSelect()
+        {
+            //Arrange - $select=Shoe,Name
+            var expected = new List<string>();
+            expected.Add("Shoe");
+            expected.Add("Name");
+
+            //Act
+            var clause = CreateSelectExpandClauseMultiSelect();
+            var actual = SelectExpandClauseExtensions.GetCurrentLevelSelectList(clause);
+
+            //Assert
+            Assert.True(expected.SequenceEqual(actual));    
+        }
+
+        [Fact]
+        public void GetCurrentLevelSelectListTestNestedSelect()
+        {
+            //Arrange - $select=FavoriteDate,Name,MyAddress/City
+            var expected = new List<string>();
+            expected.Add("FavoriteDate");
+            expected.Add("Name");
+            expected.Add("MyAddress/City");
+
+            //Act
+            var clause = CreateSelectExpandClauseNestedSelect();
+            var actual = SelectExpandClauseExtensions.GetCurrentLevelSelectList(clause);
+
+            //Assert
+            Assert.True(expected.SequenceEqual(actual));
+        }
+
+        [Fact]
+        public void GetCurrentLevelSelectListTestNestedSelectComplexType()
+        {
+            //Arrange - $select=FavoriteDate,Name,MyAddress,MyAddress/City
+            var expected = new List<string>();
+            expected.Add("FavoriteDate");
+            expected.Add("Name");
+            expected.Add("MyAddress");
+
+            //Act
+            var clause = CreateSelectExpandClauseNestedSelectWithComplexType();
+            var actual = SelectExpandClauseExtensions.GetCurrentLevelSelectList(clause);
+
+            //Assert
+            Assert.True(expected.SequenceEqual(actual));
+        }
+
+        [Fact]
+        public void GetCurrentLevelSelectListTestNestedSelectComplexTypeWithStar_FullSet()
+        {
+            //Arrange - $select=FavoriteDate,Name,MyAddress,MyAddress/City
+            var expected = new List<string>();
+            expected.Add("*");
+          
+
+            //Act
+            var clause = CreateSelectExpandClauseNestedSelectWithComplexTypeWithStar(false);
+            var actual = SelectExpandClauseExtensions.GetCurrentLevelSelectList(clause);
+
+            //Assert
+            Assert.True(expected.SequenceEqual(actual));
+        }
+
+        [Fact]
+        public void GetCurrentLevelSelectListTestNestedSelectComplexTypeWithStar_SubSet()
+        {
+            //Arrange - $select=FavoriteDate,Name,MyAddress,MyAddress/City

```

Comment:
this comment no longer matches the test -- please update the comment

Reply:
#Resolved